src/commands/dev/index.ts

```diff
@@ -531,6 +578,8 @@ export default class Dev extends BaseCommand {
     this.log('Building containers...', chalk.green('done'));
     this.log('');
 
+    await this.pruneImages(compose);
```

Comment:
Why does this run before the user's app starts and not after the user stops the compose process?

Reply:
Technically it doesn't make a difference, I just figured it would be more beneficial to clean up old images after building and before running.

If you run `architect dev` and looked and `docker images` you'd see garbage until you kill `architect dev`, which is kind of weird - doing it in between building/running means even with `architect dev` running in the background, your `docker images` list is clean. Ultimately the things we'd delete doesn't change before/after containers get run since it's all inactive/dangling image based

Reply:
(Also in the case of running `architect dev` detached, we'd want to still do this cleanup presumably and in that case would want to do it before launching containers and detaching)

Reply:
I was less concerned about the list of images than about the UX when images are being deleted. If someone had a bunch of images that needed to be deleted and that took like 20 seconds, it's way more disruptive when they're waiting for their app to start than when they kill the process

Reply:
That's fair, though I don't think it's that common of a scenario - the flip side is after killing their `architect dev` they have to wait 20 seconds to delete images which is also potentially annoying. And in general if somebody only runs `architect dev -d` they'd either never have things cleaned up (unless we deleted before running dev, like this is doing)